DB/DataObject.php

```diff
@@ -4278,7 +4304,30 @@ function free()
     function _get_table() { return $this->table(); }
     function _get_keys()  { return $this->keys();  }
 
-
+  /**
+   * Get a string to append to the query log depending on time taken.
+   *
+   * This is to allow easier grepping for slow queries.
+   *
+   * @param float $timeTaken
+   *
+   * @return string
+   */
+  public function getAlertLevel($timeTaken) {
+    if ($timeTaken >= 20) {
+      return '****';
+    }
+    if ($timeTaken > 10) {
+      return '***';
+    }
+    if ($timeTaken > 5) {
+      return '**';
+    }
+    if ($timeTaken > 1) {
```

Comment:
I think this is good to merge. My last thought is that * is actually not a great thing to grep for - as it is found in SELECT *

I thought about changing to 
~  1 sec
~~ 5 sec
~~~ 10 sec
~~~~ 20 sec 

But I'm on the fence cos it looks wierd & doesn't scream 'something is wrong'